hw/drivers/sensors/lis2dw12/src/lis2dw12.c

```diff
@@ -0,0 +1,1664 @@
+             * fail if the spi is already enabled
+             */
+            goto err;
+        }
+
+        rc = hal_spi_enable(sensor->s_itf.si_num);
+        if (rc) {
+            goto err;
+        }
+
+        rc = hal_gpio_init_out(sensor->s_itf.si_cs_pin, 1);
+        if (rc) {
+            goto err;
+        }
+    }
+
+    lis2dw12->pdd.notify_ctx.snec_sensor = sensor;
+    lis2dw12->pdd.registered_mask = 0;
+
+    rc = init_intpin(lis2dw12, lis2dw12_int_irq_handler, sensor);
+    if (rc) {
+        return rc;
+    }
+
+    return 0;
+err:
+    return rc;
+
+}
+
+/**
+ * Configure the sensor
+ *
+ * @param ptr to sensor driver
+ * @param ptr to sensor driver config
+ */
+int
+lis2dw12_config(struct lis2dw12 *lis2dw12, struct lis2dw12_cfg *cfg)
+{
+    int rc;
+    struct sensor_itf *itf;
+    uint8_t chip_id;
+    struct sensor *sensor;
+
+    itf = SENSOR_GET_ITF(&(lis2dw12->sensor));
+    sensor = &(lis2dw12->sensor);
+
+    if (itf->si_type == SENSOR_ITF_SPI) {
+
+        rc = hal_spi_disable(sensor->s_itf.si_num);
+        if (rc) {
+            goto err;
+        }
+
+        rc = hal_spi_config(sensor->s_itf.si_num, &spi_lis2dw12_settings);
+        if (rc == EINVAL) {
+            /* If spi is already enabled, for nrf52, it returns -1, We should not
+             * fail if the spi is already enabled
+             */
+            goto err;
+        }
+
+        rc = hal_spi_enable(sensor->s_itf.si_num);
+        if (rc) {
+            goto err;
+        }
+    }
+
+    rc = lis2dw12_get_chip_id(itf, &chip_id);
+    if (rc) {
+        goto err;
+    }
+
+    if (chip_id != LIS2DW12_ID) {
+        rc = SYS_EINVAL;
+        goto err;
+    }
+
+    rc = lis2dw12_reset(itf);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_write8(itf, LIS2DW12_REG_CTRL_REG2, 0x40);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_write8(itf, LIS2DW12_REG_CTRL_REG3, 0x10);
+    if (rc) {
+        goto err;
+    }
+
+    
+    rc = lis2dw12_set_offsets(itf, cfg->offset_x, cfg->offset_y, cfg->offset_z);
+    if (rc) {
+        goto err;
+    }
+    
+    rc = lis2dw12_set_filter_cfg(itf, LIS2DW12_FILTER_BW_ODR_DIV_2, 0);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_set_full_scale(itf, cfg->fs);
+    if (rc) {
+        goto err;
+    }
+
+    lis2dw12->cfg.fs = cfg->fs;
+
+    rc = lis2dw12_set_rate(itf, cfg->rate);
+    if (rc) {
+        goto err;
+    }
+
+    lis2dw12->cfg.rate = cfg->rate;
+
+    rc = lis2dw12_set_self_test(itf, LIS2DW12_ST_MODE_DISABLE);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_set_power_mode(itf, LIS2DW12_PM_HIGH_PERF);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_set_low_noise(itf, 1);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_write8(itf, LIS2DW12_REG_CTRL_REG7, 0x30);
+    if (rc) {
+        goto err;
+    }
+
+    rc = lis2dw12_set_int1_pin_cfg(itf, 0);
```

Comment:
I think the configuration for the interrupts should come from the lis2dw12_cfg which is not the case. I agree that none of it should be enabled at init time but everytime a change to the config is made, you want to be able to change interrupt config settings, which should not be 0 here.